recipes/glfw/all/patches/3.3.2-0003-Wayland-Remove-extra-cmake-modules-dependency.patch

```diff
@@ -0,0 +1,122 @@
+Date: Fri, 3 Nov 2023 08:51:04 -0500
+Subject: [PATCH] Wayland: Remove extra-cmake-modules dependency
+
+Fixes #1774.
+---
+ CMakeLists.txt     | 11 ++++----
+ src/CMakeLists.txt | 67 +++++++++++++++++++++++++++++-----------------
+ 2 files changed, 48 insertions(+), 30 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e9888624..33d1a0c1 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -237,12 +237,13 @@ endif()
+ # Use Wayland for window creation
+ #--------------------------------------------------------------------
+ if (_GLFW_WAYLAND)
+-    find_package(ECM REQUIRED NO_MODULE)
+-    list(APPEND CMAKE_MODULE_PATH "${ECM_MODULE_PATH}")
+ 
+-    find_package(Wayland REQUIRED Client Cursor Egl)
+-    find_package(WaylandScanner REQUIRED)
+-    find_package(WaylandProtocols 1.15 REQUIRED)
++    include(FindPkgConfig)
++    pkg_check_modules(Wayland REQUIRED
++        wayland-client>=0.2.7
++        wayland-cursor>=0.2.7
++        wayland-egl>=0.2.7
++        xkbcommon)
+ 
+     list(APPEND glfw_PKG_DEPS "wayland-egl")
+ 
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index e3463639..76c25251 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -30,31 +30,6 @@ elseif (_GLFW_WAYLAND)
+     set(glfw_SOURCES ${common_SOURCES} wl_init.c wl_monitor.c wl_window.c
+                      posix_time.c posix_thread.c xkb_unicode.c
+                      egl_context.c osmesa_context.c)
+-
+-    ecm_add_wayland_client_protocol(glfw_SOURCES
+-        PROTOCOL
+-        "${WAYLAND_PROTOCOLS_PKGDATADIR}/stable/xdg-shell/xdg-shell.xml"
+-        BASENAME xdg-shell)
+-    ecm_add_wayland_client_protocol(glfw_SOURCES
+-        PROTOCOL
+-        "${WAYLAND_PROTOCOLS_PKGDATADIR}/unstable/xdg-decoration/xdg-decoration-unstable-v1.xml"
+-        BASENAME xdg-decoration)
+-    ecm_add_wayland_client_protocol(glfw_SOURCES
+-        PROTOCOL
+-        "${WAYLAND_PROTOCOLS_PKGDATADIR}/stable/viewporter/viewporter.xml"
+-        BASENAME viewporter)
+-    ecm_add_wayland_client_protocol(glfw_SOURCES
+-        PROTOCOL
+-        "${WAYLAND_PROTOCOLS_PKGDATADIR}/unstable/relative-pointer/relative-pointer-unstable-v1.xml"
+-        BASENAME relative-pointer-unstable-v1)
+-    ecm_add_wayland_client_protocol(glfw_SOURCES
+-        PROTOCOL
+-        "${WAYLAND_PROTOCOLS_PKGDATADIR}/unstable/pointer-constraints/pointer-constraints-unstable-v1.xml"
+-        BASENAME pointer-constraints-unstable-v1)
+-    ecm_add_wayland_client_protocol(glfw_SOURCES
+-        PROTOCOL
+-        "${WAYLAND_PROTOCOLS_PKGDATADIR}/unstable/idle-inhibit/idle-inhibit-unstable-v1.xml"
+-        BASENAME idle-inhibit-unstable-v1)
+ elseif (_GLFW_OSMESA)
+     set(glfw_HEADERS ${common_HEADERS} null_platform.h null_joystick.h
+                      posix_time.h posix_thread.h osmesa_context.h)
+@@ -62,6 +37,48 @@ elseif (_GLFW_OSMESA)
+                      null_joystick.c posix_time.c posix_thread.c osmesa_context.c)
+ endif()
+ 
++if (_GLFW_WAYLAND)
```

Comment:
This part of the patch looks something more the applied directly to the upstream than a Conan patch.

Reply:
You're right, this fix was pulled from the upstream project, but is absolutely required to build with Wayland. I'll redo the patches based on your other comments and make it so.

Reply:
Thank you!

Reply:
I've indicated the additional patch as `upstream` instead of `portability` to try and make the patch source and scope clearer.